easybuild/tools/containers/base.py

```diff
@@ -0,0 +1,100 @@
+# #
+# Copyright 2009-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+# #
+"""
+:author: Mohamed Abidi (Bright Computing)
+"""
+import os
+
+from vsc.utils import fancylogger
+
+from easybuild.tools.build_log import EasyBuildError, print_msg
+from easybuild.tools.config import build_option, container_path
+from easybuild.tools.containers.utils import check_tool
+from easybuild.tools.filetools import write_file
+
+
+class ContainerGenerator(object):
+
+    TOOLS = {}
+
+    RECIPE_FILE_NAME = None
+
+    def __init__(self, easyconfigs):
+        self.container_base = build_option('container_base')
+        self.container_build_image = build_option('container_build_image')
+        self.container_path = container_path()
+        self.easyconfigs = easyconfigs
+        self.image_format = build_option('container_image_format')
+        self.img_name = build_option('container_image_name')
+        self.log = fancylogger.getLogger(self.__class__.__module__.replace('easybuild.', ''))
+        self.tmpdir = build_option('container_tmpdir')
+
+    def generate(self):
+        self.validate()
+        recipe_path = self.generate_recipe()
+        if self.container_build_image:
+            self.build_image(recipe_path)
+
+    def validate_tools(self):
+        for tool_name, tool_version in self.TOOLS.items():
+            if not check_tool(tool_name, tool_version):
+                raise EasyBuildError("{0} not found on your system.".format(tool_name,))
```

Comment:
Maybe the tool is there, but it's not the correct version?

Reply:
Do you suggest changing the message or being more precise about the non valid condition?

Reply:
I would change the message to also include the version requirement (if there is one)